noxfile.py

```diff
@@ -226,13 +235,77 @@ def clean(session):
     _clean(session)
 
 
+# --- Set up --- #
+
+def _setup_remotes(session):
+    """Set up the origin and upstream remote repositories."""
+    remote_cmd = ["git", "remote"]
+    posargs = list(session.posargs[1:])
+    https = "--https" in posargs
+    ssh = "--ssh" in posargs
+    username_args = extract_option_values(posargs, "--username")[0]
+    if https == ssh:
+        session.error("Exactly one of '--https' or '--ssh' must be passed")
+
+    # Get current origin details
+    origin_url_cmd = (*remote_cmd, "get-url", "origin")
+    origin_url = session.run(
+        *origin_url_cmd, external=True, silent=True, log=False).strip()
+    if "https://" not in origin_url:
+        origin_url = origin_url.split(":")[-1]
+    origin_user, origin_repo = origin_url.split("/")[-2:]
+    if origin_repo.endswith(".git"):
+        origin_repo = origin_repo[:-4]
+
+    # Check username
+    if username_args:
+        origin_user = username_args[0].strip().lstrip("@")
+    elif origin_user.lower() == ORG_NAME.lower():
+        code_host = REPO_URL_HTTPS.split(":")[1].lstrip("/").split("/")[0]
+        session.warn(
+            "Origin remote currently set to upstream; should be your fork.\n"
+            f"To fix, fork it and pass --username <Your {code_host} username>"
+        )
+
+    # Set up remotes
+    existing_remotes = session.run(
+        *remote_cmd, external=True, silent=True, log=False).strip().split("\n")
+    for remote, user_name, repo_name in (
+            ("origin", origin_user, origin_repo),
+            ("upstream", ORG_NAME, REPO_NAME),
```

Comment:
```suggestion
        ("origin", origin_user, origin_repo),
        ("upstream", ORG_NAME, REPO_NAME),
```

I think we don't need two levels of indentation here. I mean, Black/Ruff will probably format it with a single level.

Reply:
Good point; I thought this was Black style and we were running Black automatically on this like some of our other repos but evidently not.

Since there are plenty of other small departures from Black style and its not directly related to this PR, I recommend we just merge this and I'll drop a followup PR adding Black to the pre-commit hooks and blackening the relevant files to permanently solve this problem.

Reply:
Ok, no prob.